server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -879,35 +886,54 @@ public OwnerDTO createOwner(@ApiParam(name = "owner", required = true) OwnerDTO
             throw new BadRequestException(i18n.tr("Owners must be created with a valid key"));
         }
 
-        // Validate and set content access mode list & content access mode
-        if (StringUtils.isBlank(dto.getContentAccessModeList())) {
-            dto.setContentAccessModeList(ContentAccessCertServiceAdapter.DEFAULT_CONTENT_ACCESS_MODE);
-        }
+        Owner owner = new Owner();
+        owner.setKey(dto.getKey());
 
-        if (StringUtils.isBlank(dto.getContentAccessMode())) {
-            dto.setContentAccessMode(ContentAccessCertServiceAdapter.DEFAULT_CONTENT_ACCESS_MODE);
+        // Check that the default service level is *not* set at this point
+        if (!StringUtils.isBlank(dto.getDefaultServiceLevel())) {
+            throw new BadRequestException(
+                i18n.tr("The default service level cannot be specified during owner creation"));
         }
 
-        if (!containsContentAccessMode(dto.getContentAccessModeList(), dto.getContentAccessMode())) {
-            throw new BadRequestException(
-                i18n.tr("The content access mode \"{1}\" is not allowed for this owner.",
-                    dto.getContentAccessMode()));
+        // Validate and set content access mode list & content access mode
+        boolean configureContentAccess = false;
+
+        final String defaultContentAccess = ContentAccessMode.getDefault().toDatabaseValue();
+        String contentAccessModeList = dto.getContentAccessModeList();
+        String contentAccessMode = dto.getContentAccessMode();
+
+        if (!StringUtils.isBlank(contentAccessMode) && !contentAccessMode.equals(defaultContentAccess)) {
+            if (config.getBoolean(ConfigProperties.STANDALONE)) {
+                throw new BadRequestException(
+                    i18n.tr("The owner content access mode and content access mode list cannot be set " +
+                        "directly in standalone mode."));
```

Comment:
I see this was not a restriction on the create owner call before (both the mode and the list could be set). Was that an omission? Also, I assume this has something to do with the implementation note below about pre-existing expected behaviours; what are those?

Also, if we're adding this restriction here now, we should add a test case about it (best place is probably OwnerResourceTest.java)

Reply:
It looked to be an omission, and from the bit of discussion I had with people who remembered the original design, it seemed like an oversight. Prior, we could set the content access mode and list during creation, but it would not result in an actual change in state or cert generation at all. So you could get into a situation where an owner was already in org_environment but it wouldn't be functional, and the only way to fix it would be to do two updates to flip to entitlement and then back to org_environment so the updateOwnerContentAccess call (previously refreshContentAccess) would get hit.

This got added to prevent that from happening, but I quickly found that we have a lot of behavior that's expecting some parts of this to be allowed -- i.e. explicitly setting the default values during creation, and specifying the existing values during update (such as when a client fetches an org, makes changes to the received json and then sends it back up).

There were a few tests which broke by adding this restriction, and those were updated or skipped in standalone mode where it either didn't make sense or didn't seem to be reconcilable.